runtime/onert/backend/train/MemoryPlanner.h

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file        MemoryPlanner.h
+ * @brief       This file contains Memory Planning related classes
+ */
+
+#ifndef __ONERT_BACKEND_TRAIN_MEMORY_PLANNER_H__
+#define __ONERT_BACKEND_TRAIN_MEMORY_PLANNER_H__
+
+#include <cassert>
+#include <map>
+#include <vector>
+#include <unordered_set>
+#include <memory>
+
+#include <backend/basic/Allocator.h>
+#include <backend/basic/IMemoryPlanner.h>
+#include "util/logging.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace train
+{
+
+/**
+ * @brief Class to plan memory by bump way
+ */
+template <typename Index> class BumpPlanner : public basic::IMemoryPlanner<Index>
```

Comment:
Do we need `template <typename Index>` ?

Reply:
I guess the other temporary tensors may need another index(key), but I'm not sure it.

Reply:
IMO, we can change this to use template when we need, not now.